File: netDxf/Entities/MLeaderArrowHead.Fields.cs
// Copyright (c) netDxf contributors. Licensed under the MIT License.
using System;
using netDxf.Blocks;
using netDxf.Objects;
using netDxf.Tables;

namespace netDxf.Entities
{
    /// <summary>Editable stored MLeaderArrowHead fields. Redundant values are retained independently; angles are radians and colors are DXF raw-color integers.</summary>
    public sealed partial class MLeaderArrowHead : MLeaderData
    {
        private static readonly MLeaderField[] fields = new MLeaderField[]
        {
            new MLeaderField(94, typeof(int), 0, false, 2007),
            new MLeaderField(345, typeof(BlockRecord), null, true, 2007),
        };
        internal override MLeaderField[] Fields { get { return fields; } }
        /// <summary>Gets or sets the stored group 94 value.</summary>
        public int Index { get { return this.Get<int>(94); } set { this.Set(94, value); } }
        /// <summary>Gets or sets the registered object reference; null is the absent DXF handle.</summary>
        public BlockRecord Block { get { return this.Get<BlockRecord>(345); } set { this.Set(345, value); } }
    }
}

File: javascript/netDxf/Entities/MLeaderArrowHead.Fields.js
// Copyright (c) netDxf contributors. MIT License; see package LICENSE.
import {MLeaderData,InstallMLeaderFields,MLeaderComponentTypes} from './MLeaderData.js';
import {InitializeMLeaderChildren,InstallMLeaderChildren} from './MLeaderContext.js';
import {Vector3} from '../Vector3.js';
import {BlockRecord} from '../Blocks/BlockRecord.js';
export class MLeaderArrowHead extends MLeaderData {
  constructor(){super();InitializeMLeaderChildren(this,'MLeaderArrowHead');}
}
InstallMLeaderFields(MLeaderArrowHead,[
  ['Index',94,'int',0,false,2007],
  ['Block',345,()=>BlockRecord,null,true,2007],
]);
MLeaderComponentTypes.set('MLeaderArrowHead',MLeaderArrowHead);
InstallMLeaderChildren(MLeaderArrowHead,'MLeaderArrowHead');
